python/lsst/obs/decam/ingestCalibs.py

```diff
@@ -0,0 +1,54 @@
+from lsst.pipe.tasks.ingestCalibs import CalibsParseTask
+
+class DecamCalibsParseTask(CalibsParseTask):
+    def getInfo(self, filename):
+        """Get information about the image from the filename and/or its contents
+
+        @param filename: Name of file to inspect
+        @return File properties; list of file properties for each extension
+        """
+        phuInfo, infoList = CalibsParseTask.getInfo(self, filename)
+        # Single-extension fits without EXTNAME can be a valid CP calibration product
+        # Use info of primary header unit
+        if not infoList:
+            infoList.append(phuInfo)
+        return phuInfo, infoList
+
+    def translate_ccdnum(self, md):
+        """Return CCDNUM as a integer
+
+        @param md: FITS header metadata
+        """
+        if md.exists("CCDNUM"):
+            ccdnum = md.get("CCDNUM")
+        else:
+            self.log.warn("Unable to find value for CCDNUM")
+            ccdnum = None
+        # Some MasterCal from NOAO Archive has 2 CCDNUM keys in each HDU
+        # Make sure only one integer is returned. 
+        if not isinstance(ccdnum, int) and isinstance(ccdnum, (tuple, list)):
+            try:
+                ccdnum = ccdnum[0]
+            except KeyError:
+                ccdnum = None
+        return ccdnum
+
+    def translate_date(self, md):
+        """Convert a full DATE-OBS to a mere date or return unknown
+
+        @param md: FITS header metadata
+        """
+        if md.exists("DATE-OBS"):
+            date = md.get("DATE-OBS").strip()
+            c = date.find("T")
+            if c > 0:
+                date = date[:c]
+        else:
+            self.log.warn("Unable to find value for DATE-OBS")
+            date = "unknown"
+        return date
```

Comment:
so if the DATE-OBS header does not contain a "T" for some reason this function just returns whatever was present in the header? If there is no T the header is not compliant so I'd rather the routine got upset. If you are trying to support pre-Y2K standard DATE-OBS headers then there will be inconsistencies because those used "/" rather than "-".

I think it would be much safer if this code used a pattern match of something like `^\d\d\d\d-\d\d-\d\dT' so it could be sure what is actually being returned.